extension/src/data/index.ts

```diff
@@ -9,8 +9,15 @@ import { uniqueValues } from '../util/array'
 import { DeferredDisposable } from '../class/deferred'
 import { isSameOrChild } from '../fileSystem'
 
+export type ExperimentsOutput = {
+  currentBranch: string
+  expShow: ExpShowOutput
+  rowOrder: { branch: string; sha: string }[]
```

Comment:
[F] `rowOrder` is the key to duplication between branches. It is created from making multiple calls to `git log`. We no longer duplicate the commit data anywhere. We only send "duplicate" records to the experiments table. I have even sorted the commits by `Created` so they are displayed chronologically in the tree.